src/test/java/jp/co/yahoo/yosegi/message/formatter/json/TestJacksonMessageWriter.java

```diff
@@ -0,0 +1,288 @@
+      this.isArray = isArray;
+    }
+
+    @Override
+    public PrimitiveObject get(final String key ) throws IOException {
+      if ( "key1".equals( key ) ) {
+        return new StringObj( "a" );
+      } else if ( "key2".equals( key ) ) {
+        return new StringObj( "b" );
+      } else if ( "key3".equals( key ) ) {
+        return new StringObj( "c" );
+      }
+      return null;
+    }
+
+    @Override
+    public PrimitiveObject get( final int index ) throws IOException {
+      if ( index == 0 ) {
+        return new StringObj( "a" );
+      } else if ( index == 1 ) {
+        return new StringObj( "b" );
+      } else if ( index == 2 ) {
+        return new StringObj( "c" );
+      }
+      return null;
+    }
+
+    @Override
+    public IParser getParser( final String key ) throws IOException {
+      return new JacksonNullParser();
+    }
+
+    @Override
+    public IParser getParser( final int index ) throws IOException {
+      return new JacksonNullParser();
+    }
+
+    @Override
+    public String[] getAllKey() throws IOException {
+      return new String[]{ "key1" , "key2" , "key3" };
+    }
+
+    @Override
+    public boolean containsKey( final String key ) throws IOException {
+      if ( "key1".equals( key ) ) {
+        return true;
+      } else if ( "key2".equals( key ) ) {
+        return true;
+      } else if ( "key3".equals( key ) ) {
+        return true;
+      }
+      return false;
+    }
+
+    @Override
+    public int size() throws IOException {
+      return 3;
+    }
+
+    @Override
+    public boolean isArray() throws IOException {
+      return isArray;
+    }
+
+    @Override
+    public boolean isMap() throws IOException {
+      return ! isArray;
+    }
+
+    @Override
+    public boolean isStruct() throws IOException {
+      return ! isArray;
+    }
+
+    @Override
+    public boolean hasParser( final int index ) throws IOException {
+      return false;
+    }
+
+    @Override
+    public boolean hasParser( final String key ) throws IOException {
+      return false;
+    }
+
+    @Override
+    public Object toJavaObject() throws IOException {
+      return null;
+    }
+
+  }
+
+  @Test
+  public void T_createObj_1() throws IOException {
+    ObjectMapper mapper = new ObjectMapper();
+    JacksonMessageWriter writer = new JacksonMessageWriter();
+    byte[] result = writer.create( new StringObj( "a" ) );
+    JsonNode node = mapper.readTree( result );
+    assertTrue( ( node instanceof TextNode ) );
+    assertEquals( node.asText() , "a" );
+  }
+
+  @Test
+  public void T_createObj_2() throws IOException {
+    ObjectMapper mapper = new ObjectMapper();
+    JacksonMessageWriter writer = new JacksonMessageWriter();
+    PrimitiveObject obj = null;
+    byte[] result = writer.create( obj );
+    JsonNode node = mapper.readTree( result );
+    assertTrue( ( node instanceof NullNode ) );
+  }
+
+  @Test
+  public void T_createArray_1() throws IOException {
```

Comment:
Please write the test method name is to describe what is wanted to in the test instead of using number (it makes vague one).

Reply:
I changed the method name.